packages/backend-common/src/scm/git.ts

```diff
@@ -90,7 +93,8 @@ export class Git {
       dir,
       ref,
       singleBranch: true,
-      depth: 1,
+      depth: depth ?? 1,
+      noCheckout: noCheckout ?? false,
```

Comment:
```suggestion
      noCheckout,
```

This should work as well and would remove the need to assert for noCheckout false in the test above.